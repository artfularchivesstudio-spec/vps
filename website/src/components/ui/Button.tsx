import React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';
import { clsx } from 'clsx';

const buttonVariants = cva(
  'inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2',
  {
    variants: {
      variant: {
        default:
          'bg-primary text-white hover:bg-primary/90 focus:ring-primary',
        destructive:
          'bg-danger text-white hover:bg-danger/90 focus:ring-danger',
        outline:
          'border border-input bg-transparent shadow-sm hover:bg-accent hover:text-accent-foreground focus:ring-accent',
        secondary:
          'bg-secondary text-white hover:bg-secondary/80 focus:ring-secondary',
        ghost: 'hover:bg-accent hover:text-accent-foreground focus:ring-accent',
        link: 'text-primary underline-offset-4 hover:underline focus:ring-primary',
      },
      size: {
        default: 'h-10 px-4 py-2',
        sm: 'h-9 rounded-md px-3',
        lg: 'h-11 rounded-md px-8',
        icon: 'h-10 w-10',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
    },
  }
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, ...props }, ref) => {
    return (
      <button
        className={clsx(buttonVariants({ variant, size, className }))}
        ref={ref}
        {...props}
      />
    );
  }
);
Button.displayName = 'Button';

export { Button, buttonVariants };