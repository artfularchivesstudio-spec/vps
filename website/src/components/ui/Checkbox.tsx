import { cn } from '@/lib/utils'
import * as React from 'react'

const Checkbox = React.forwardRef<
  HTMLInputElement,
  React.InputHTMLAttributes<HTMLInputElement> & {
    indeterminate?: boolean
  }
>(({ className, indeterminate, ...props }, ref) => {
  const inputRef = React.useRef<HTMLInputElement>(null)
  
  React.useEffect(() => {
    if (inputRef.current) {
      inputRef.current.indeterminate = indeterminate || false
    }
  }, [indeterminate])

  React.useImperativeHandle(ref, () => inputRef.current!)

  return (
    <input
      ref={inputRef}
      type="checkbox"
      className={cn(
        'h-4 w-4 rounded border-gray-300 text-indigo-600 focus:ring-indigo-500 dark:border-gray-600 dark:bg-gray-800',
        'focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500',
        'disabled:cursor-not-allowed disabled:opacity-50',
        className
      )}
      {...props}
    />
  )
})
Checkbox.displayName = 'Checkbox'

export { Checkbox }
