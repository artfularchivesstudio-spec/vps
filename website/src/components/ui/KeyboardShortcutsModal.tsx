'use client'
import React from 'react';

import { motion, AnimatePresence } from 'framer-motion'

interface KeyboardShortcut {
  key: string
  ctrlKey?: boolean
  metaKey?: boolean
  shiftKey?: boolean
  altKey?: boolean
  action: () => void
  description: string
  category?: string
}

interface KeyboardShortcutsModalProps {
  isOpen: boolean
  onClose: () => void
  groupedShortcuts: Record<string, KeyboardShortcut[]>
  formatShortcut: (shortcut: KeyboardShortcut) => string
}

export default function KeyboardShortcutsModal({
  isOpen,
  onClose,
  groupedShortcuts,
  formatShortcut
}: KeyboardShortcutsModalProps) {
  return (
    <AnimatePresence>
      {isOpen && (
        <motion.div
          className="fixed inset-0 bg-black/60 backdrop-blur-sm z-50 flex items-center justify-center p-4"
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          onClick={onClose}
        >
          <motion.div
            className="bg-white rounded-2xl shadow-2xl max-w-2xl w-full max-h-[80vh] overflow-hidden"
            initial={{ scale: 0.9, opacity: 0 }}
            animate={{ scale: 1, opacity: 1 }}
            exit={{ scale: 0.9, opacity: 0 }}
            onClick={(e) => e.stopPropagation()}
          >
            {/* Header */}
            <div className="bg-gradient-to-r from-purple-600 to-pink-600 text-white p-6">
              <div className="flex items-center justify-between">
                <div>
                  <h2 className="text-2xl font-bold">‚å®Ô∏è Keyboard Shortcuts</h2>
                  <p className="text-purple-100 mt-1">Speed up your workflow with these shortcuts</p>
                </div>
                <button
                  onClick={onClose}
                  className="text-white hover:text-purple-200 transition-colors"
                >
                  <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M6 18L18 6M6 6l12 12" />
                  </svg>
                </button>
              </div>
            </div>

            {/* Content */}
            <div className="p-6 overflow-y-auto max-h-[calc(80vh-120px)]">
              <div className="space-y-6">
                {Object.entries(groupedShortcuts).map(([category, shortcuts]) => (
                  <div key={category}>
                    <h3 className="text-lg font-semibold text-gray-800 mb-3 flex items-center gap-2">
                      <span className="w-2 h-2 bg-gradient-to-r from-purple-500 to-pink-500 rounded-full"></span>
                      {category}
                    </h3>
                    <div className="space-y-2">
                      {shortcuts.map((shortcut, index) => (
                        <div
                          key={`${category}-${index}`}
                          className="flex items-center justify-between py-2 px-3 rounded-lg hover:bg-gray-50 transition-colors"
                        >
                          <span className="text-gray-700">{shortcut.description}</span>
                          <kbd className="px-3 py-1 bg-gray-100 border border-gray-300 rounded-md text-sm font-mono text-gray-800 shadow-sm">
                            {formatShortcut(shortcut)}
                          </kbd>
                        </div>
                      ))}
                    </div>
                  </div>
                ))}
              </div>

              {/* Pro Tips */}
              <div className="mt-8 p-4 bg-gradient-to-r from-blue-50 to-purple-50 rounded-lg border border-blue-200">
                <h4 className="font-semibold text-blue-800 mb-2">üí° Pro Tips</h4>
                <ul className="text-sm text-blue-700 space-y-1">
                  <li>‚Ä¢ Shortcuts work everywhere except when typing in text fields</li>
                  <li>‚Ä¢ Use <kbd className="px-1 bg-blue-100 rounded text-xs">Ctrl+S</kbd> or <kbd className="px-1 bg-blue-100 rounded text-xs">‚åò+S</kbd> to save even while typing</li>
                  <li>‚Ä¢ Press <kbd className="px-1 bg-blue-100 rounded text-xs">?</kbd> anytime to see this help</li>
                </ul>
              </div>
            </div>
          </motion.div>
        </motion.div>
      )}
    </AnimatePresence>
  )
}