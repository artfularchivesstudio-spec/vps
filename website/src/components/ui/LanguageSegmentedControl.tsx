/**
 * 🎭 The Language Segmented Control - A Symphony of Multilingual Navigation
 *
 * "Like elegant piano keys in a linguistic concert hall, each language button 
 * awaits its moment to perform, while the selected language glows with the 
 * spotlight of active engagement. A theatrical toggle that transforms 
 * the reading experience into a global performance."
 *
 * - The Interface Maestro
 */

'use client'

import { motion, AnimatePresence } from 'framer-motion'
import { useState, useEffect } from 'react'

export interface Language {
  code: string
  name: string
  flag: string
  available: boolean
}

interface LanguageSegmentedControlProps {
  languages: Language[]
  activeLanguage: string
  onChange: (languageCode: string) => void
  className?: string
  showLabels?: boolean
  size?: 'sm' | 'md' | 'lg'
}

export default function LanguageSegmentedControl({
  languages,
  activeLanguage,
  onChange,
  className = '',
  showLabels = true,
  size = 'md'
}: LanguageSegmentedControlProps) {
  const [hoveredLanguage, setHoveredLanguage] = useState<string | null>(null)

  // 🎨 The size variations - different theatrical stages
  const sizeClasses = {
    sm: {
      container: 'h-8 text-xs',
      button: 'px-3 py-1',
      flag: 'text-sm',
      label: 'text-xs'
    },
    md: {
      container: 'h-10 text-sm',
      button: 'px-4 py-2',
      flag: 'text-base',
      label: 'text-sm'
    },
    lg: {
      container: 'h-12 text-base',
      button: 'px-6 py-3',
      flag: 'text-lg',
      label: 'text-base'
    }
  }

  const sizeClass = sizeClasses[size]
  
  // 🌈 The color palette for our linguistic performance
  const colorVariants = {
    en: { bg: 'bg-blue-500', text: 'text-blue-600', hover: 'hover:bg-blue-50' },
    es: { bg: 'bg-red-500', text: 'text-red-600', hover: 'hover:bg-red-50' },
    hi: { bg: 'bg-orange-500', text: 'text-orange-600', hover: 'hover:bg-orange-50' }
  }

  // 🎪 Filter only available languages for display
  const availableLanguages = languages.filter(lang => lang.available)

  if (availableLanguages.length <= 1) {
    // 🎭 If only one language is available, show a simple indicator instead
    return (
      <div className={`flex items-center gap-2 ${className}`}>
        <span className="text-2xl">{availableLanguages[0]?.flag || '🇺🇸'}</span>
        {showLabels && (
          <span className="text-sm text-gray-600">
            {availableLanguages[0]?.name || 'English'}
          </span>
        )}
      </div>
    )
  }

  return (
    <div className={`relative ${className}`}>
      {/* 🎭 The Main Stage - Segmented Control Container */}
      <div className={`
        relative flex items-center bg-gray-100 rounded-full p-1 ${sizeClass.container}
        shadow-inner border border-gray-200
      `}>
        {/* 🌟 The Spotlight - Moving Background for Active Language */}
        <motion.div
          className="absolute top-1 bottom-1 bg-white rounded-full shadow-md border border-gray-300/50"
          initial={false}
          animate={{
            left: `${(availableLanguages.findIndex(lang => lang.code === activeLanguage) * (100 / availableLanguages.length)) + 1}%`,
            width: `${(100 / availableLanguages.length) - 2}%`
          }}
          transition={{ type: "spring", stiffness: 400, damping: 30 }}
        />

        {/* 🎪 The Language Buttons - Our Performing Cast */}
        {availableLanguages.map((language, index) => {
          const isActive = language.code === activeLanguage
          const isHovered = hoveredLanguage === language.code
          const colors = colorVariants[language.code as keyof typeof colorVariants] || colorVariants.en

          return (
            <motion.button
              key={language.code}
              onClick={() => onChange(language.code)}
              onMouseEnter={() => setHoveredLanguage(language.code)}
              onMouseLeave={() => setHoveredLanguage(null)}
              className={`
                relative z-10 flex-1 flex items-center justify-center gap-1.5 
                ${sizeClass.button} rounded-full font-medium transition-all duration-200
                ${isActive 
                  ? `${colors.text} shadow-sm` 
                  : `text-gray-600 ${colors.hover}`
                }
                ${isHovered && !isActive ? 'scale-105' : ''}
              `}
              whileHover={{ scale: isActive ? 1 : 1.05 }}
              whileTap={{ scale: 0.95 }}
            >
              {/* 🏁 The Flag - National Pride in Emoji Form */}
              <motion.span 
                className={`${sizeClass.flag}`}
                animate={{ 
                  rotate: isHovered ? [0, -10, 10, -5, 0] : 0,
                  scale: isActive ? 1.1 : 1
                }}
                transition={{ duration: 0.3 }}
              >
                {language.flag}
              </motion.span>

              {/* 📝 The Label - Linguistic Identity */}
              {showLabels && (
                <motion.span
                  className={`${sizeClass.label} font-medium hidden sm:block`}
                  animate={{ 
                    fontWeight: isActive ? 600 : 500,
                    opacity: isActive ? 1 : 0.8 
                  }}
                >
                  {language.name}
                </motion.span>
              )}

              {/* ✨ The Active Indicator - A Gentle Glow */}
              {isActive && (
                <motion.div
                  className={`absolute inset-0 rounded-full ${colors.bg} opacity-10`}
                  initial={{ scale: 0.8, opacity: 0 }}
                  animate={{ scale: 1, opacity: 0.1 }}
                  exit={{ scale: 0.8, opacity: 0 }}
                />
              )}
            </motion.button>
          )
        })}
      </div>

      {/* 🎨 The Tooltip - Hovering Wisdom */}
      <AnimatePresence>
        {hoveredLanguage && hoveredLanguage !== activeLanguage && (
          <motion.div
            className="absolute top-full left-1/2 transform -translate-x-1/2 mt-2 z-20"
            initial={{ opacity: 0, y: -5 }}
            animate={{ opacity: 1, y: 0 }}
            exit={{ opacity: 0, y: -5 }}
            transition={{ duration: 0.2 }}
          >
            <div className="bg-gray-900 text-white text-xs px-3 py-1 rounded-md shadow-lg">
              <div className="flex items-center gap-1">
                <span>{languages.find(l => l.code === hoveredLanguage)?.flag}</span>
                <span>Switch to {languages.find(l => l.code === hoveredLanguage)?.name}</span>
              </div>
              {/* 📐 The Tooltip Arrow */}
              <div className="absolute bottom-full left-1/2 transform -translate-x-1/2">
                <div className="w-0 h-0 border-l-4 border-r-4 border-b-4 border-transparent border-b-gray-900"></div>
              </div>
            </div>
          </motion.div>
        )}
      </AnimatePresence>
    </div>
  )
}

