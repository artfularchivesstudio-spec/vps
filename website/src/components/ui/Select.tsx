import { cn } from '@/lib/utils'
import * as React from 'react'

const Select = React.forwardRef<
  HTMLSelectElement,
  React.SelectHTMLAttributes<HTMLSelectElement>
>(({ className, children, ...props }, ref) => {
  return (
    <select
      ref={ref}
      className={cn(
        'w-full px-3 py-2 border border-gray-300 rounded-md focus:outline-none focus:ring-2 focus:ring-indigo-500 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-100',
        className
      )}
      {...props}
    >
      {children}
    </select>
  )
})
Select.displayName = 'Select'

const SelectContent = ({ children }: { children: React.ReactNode }) => {
  return <>{children}</>
}

const SelectItem = ({ children, value }: { children: React.ReactNode; value: string }) => {
  return <option value={value}>{children}</option>
}

const SelectTrigger = ({ children }: { children: React.ReactNode }) => {
  return <>{children}</>
}

const SelectValue = ({ children }: { children: React.ReactNode }) => {
  return <>{children}</>
}

export { Select, SelectContent, SelectItem, SelectTrigger, SelectValue }
