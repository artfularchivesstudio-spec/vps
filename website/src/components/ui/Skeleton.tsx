import { cn } from '@/lib/utils'

interface SkeletonProps {
  className?: string
  lines?: number
  height?: string
  width?: string
}

export const Skeleton = ({ className, lines = 1, height = 'h-4', width = 'w-full' }: SkeletonProps) => {
  return (
    <div className="space-y-2">
      {Array.from({ length: lines }).map((_, i) => (
        <div
          key={i}
          className={cn(
            'animate-pulse rounded-md bg-gray-200 dark:bg-gray-700',
            height,
            width,
            className
          )}
        />
      ))}
    </div>
  )
}

export const SkeletonCard = ({ className }: { className?: string }) => {
  return (
    <div className={cn('animate-pulse', className)}>
      <div className="rounded-md bg-gray-200 dark:bg-gray-700 h-48 mb-4" />
      <Skeleton lines={3} />
    </div>
  )
}

export const SkeletonTable = ({ rows = 5, cols = 4 }: { rows?: number; cols?: number }) => {
  return (
    <div className="space-y-2">
      {/* Header */}
      <div className="grid gap-4" style={{ gridTemplateColumns: `repeat(${cols}, 1fr)` }}>
        {Array.from({ length: cols }).map((_, i) => (
          <div key={i} className="h-6 bg-gray-200 dark:bg-gray-700 rounded animate-pulse" />
        ))}
      </div>
      {/* Rows */}
      {Array.from({ length: rows }).map((_, rowIndex) => (
        <div key={rowIndex} className="grid gap-4" style={{ gridTemplateColumns: `repeat(${cols}, 1fr)` }}>
          {Array.from({ length: cols }).map((_, colIndex) => (
            <div key={colIndex} className="h-10 bg-gray-100 dark:bg-gray-800 rounded animate-pulse" />
          ))}
        </div>
      ))}
    </div>
  )
}