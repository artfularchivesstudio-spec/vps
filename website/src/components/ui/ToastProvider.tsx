'use client'

import { cn } from '@/lib/utils'
import { AlertTriangle, CheckCircle, Info, X, XCircle } from 'lucide-react'
import { createContext, ReactNode, useCallback, useContext, useState } from 'react'

interface Toast {
  id: string
  emoji: string
  title: string
  description?: string
  type: 'success' | 'error' | 'warning' | 'info'
  durationMs?: number
}

interface ToastContextType {
  push: (toast: Omit<Toast, 'id'>) => void
}

export const ToastContext = createContext<ToastContextType | undefined>(undefined)

export const useToast = () => {
  const context = useContext(ToastContext)
  if (!context) {
    throw new Error('useToast must be used within a ToastProvider')
  }
  return context
}

interface ToastProviderProps {
  children: ReactNode
}

export const ToastProvider = ({ children }: ToastProviderProps) => {
  const [toasts, setToasts] = useState<Toast[]>([])

  const dismiss = useCallback((id: string) => {
    setToasts(prev => prev.filter(toast => toast.id !== id))
  }, [])

  const push = useCallback((toast: Omit<Toast, 'id'>) => {
    const id = Math.random().toString(36).substr(2, 9)
    const newToast: Toast = {
      ...toast,
      id,
      durationMs: toast.durationMs || 5000,
    }
    
    setToasts(prev => [...prev, newToast])

    // Auto-dismiss after duration
    setTimeout(() => {
      dismiss(id)
    }, newToast.durationMs)
  }, [dismiss])

  const getIcon = (type: Toast['type']) => {
    switch (type) {
      case 'success':
        return <CheckCircle className="h-5 w-5 text-green-500" />
      case 'error':
        return <XCircle className="h-5 w-5 text-red-500" />
      case 'warning':
        return <AlertTriangle className="h-5 w-5 text-yellow-500" />
      case 'info':
        return <Info className="h-5 w-5 text-blue-500" />
    }
  }

  const getBgColor = (type: Toast['type']) => {
    switch (type) {
      case 'success':
        return 'bg-green-50 dark:bg-green-900/20 border-green-200 dark:border-green-800'
      case 'error':
        return 'bg-red-50 dark:bg-red-900/20 border-red-200 dark:border-red-800'
      case 'warning':
        return 'bg-yellow-50 dark:bg-yellow-900/20 border-yellow-200 dark:border-yellow-800'
      case 'info':
        return 'bg-blue-50 dark:bg-blue-900/20 border-blue-200 dark:border-blue-800'
    }
  }

  return (
    <ToastContext.Provider value={{ push }}>
      {children}
      
      {/* Toast Container */}
      <div className="fixed top-4 right-4 z-50 space-y-2 max-w-sm">
        {toasts.map((toast) => (
          <div
            key={toast.id}
            className={cn(
              'flex items-start p-4 rounded-lg border shadow-lg transition-all duration-300',
              getBgColor(toast.type),
              'animate-in slide-in-from-right-5 fade-in-0'
            )}
          >
            <div className="flex-shrink-0">
              {getIcon(toast.type)}
            </div>
            <div className="ml-3 flex-1">
              <div className="flex items-center">
                <span className="text-lg mr-2">{toast.emoji}</span>
                <h3 className="text-sm font-medium text-gray-900 dark:text-gray-100">
                  {toast.title}
                </h3>
              </div>
              {toast.description && (
                <p className="mt-1 text-sm text-gray-600 dark:text-gray-400">
                  {toast.description}
                </p>
              )}
            </div>
            <button
              onClick={() => dismiss(toast.id)}
              className="ml-4 flex-shrink-0 rounded-md p-1 inline-flex text-gray-400 hover:text-gray-600 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              <X className="h-4 w-4" />
            </button>
          </div>
        ))}
      </div>
    </ToastContext.Provider>
  )
}
