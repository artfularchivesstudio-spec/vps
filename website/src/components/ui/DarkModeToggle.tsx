'use client'

import { Button } from '@/components/ui/Button'
import { Moon, Sun } from 'lucide-react'
import { useEffect, useState } from 'react'

export const DarkModeToggle = () => {
  const [isDark, setIsDark] = useState(false)

  useEffect(() => {
    // Check for saved theme preference or default to light mode
    const savedTheme = localStorage.getItem('theme')
    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches
    
    const shouldBeDark = savedTheme === 'dark' || (!savedTheme && prefersDark)
    
    if (shouldBeDark) {
      document.documentElement.classList.add('dark')
      setIsDark(true)
    } else {
      document.documentElement.classList.remove('dark')
      setIsDark(false)
    }
  }, [])

  const toggleDarkMode = () => {
    const newIsDark = !isDark
    setIsDark(newIsDark)
    
    if (newIsDark) {
      document.documentElement.classList.add('dark')
      localStorage.setItem('theme', 'dark')
    } else {
      document.documentElement.classList.remove('dark')
      localStorage.setItem('theme', 'light')
    }
  }

  return (
    <Button
      variant="ghost"
      size="sm"
      onClick={toggleDarkMode}
      className="relative"
      aria-label={isDark ? 'Switch to light mode' : 'Switch to dark mode'}
    >
      <Sun className={`h-4 w-4 transition-all ${isDark ? 'rotate-90 scale-0' : 'rotate-0 scale-100'}`} />
      <Moon className={`absolute h-4 w-4 transition-all ${isDark ? 'rotate-0 scale-100' : 'rotate-90 scale-0'}`} />
    </Button>
  )
}