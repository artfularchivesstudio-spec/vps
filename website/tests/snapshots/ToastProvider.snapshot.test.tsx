import React from 'react'
import { render } from '@testing-library/react'
import { ToastProvider, useToast } from '../../src/components/ui/ToastProvider'

function Demo() {
  const { push } = useToast()
  React.useEffect(() => {
    push({ emoji: 'âœ…', title: 'Saved', description: 'Post saved successfully', type: 'success', durationMs: 0 })
  }, [push])
  return null
}

describe('ToastProvider snapshots', () => {
  it('renders a toast', () => {
    const { asFragment, container } = render(
      <ToastProvider>
        <Demo />
      </ToastProvider>
    )
    // Basic assertion without jest-dom
    if (!container.textContent?.includes('Saved')) {
      throw new Error('Toast not rendered')
    }
    expect(asFragment()).toMatchSnapshot()
  })
})
