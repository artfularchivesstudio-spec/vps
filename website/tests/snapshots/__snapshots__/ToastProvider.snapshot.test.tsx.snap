// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToastProvider snapshots > renders a toast 1`] = `
<DocumentFragment>
  <div
    class="fixed top-4 right-4 z-[1000] space-y-2"
  >
    <div
      class="pointer-events-auto max-w-sm w-80 rounded-lg shadow-lg border p-3 bg-white/95 backdrop-blur-sm flex items-start space-x-2 border-green-200"
    >
      <div
        class="text-lg mt-0.5"
      >
        ✅
      </div>
      <div
        class="flex-1"
      >
        <div
          class="text-sm font-medium text-gray-900"
        >
          Saved
        </div>
        <div
          class="text-xs text-gray-600 mt-0.5"
        >
          Post saved successfully
        </div>
      </div>
      <button
        class="text-xs text-gray-500 hover:text-gray-700"
      >
        ✕
      </button>
    </div>
  </div>
</DocumentFragment>
`;
